uvicorn/supervisors/multiprocess.py

```diff
@@ -23,11 +23,18 @@ def __init__(self, config, target, sockets):
         self.processes = []
         self.should_exit = threading.Event()
         self.pid = os.getpid()
+        self.child_pids = []
 
     def signal_handler(self, sig, frame):
         """
         A signal handler that is registered with the parent process.
         """
+        for child_pid in self.child_pids:
+            try:
+                os.kill(child_pid, signal.SIGINT)
+                finished = os.waitpid(child_pid, 0)
```

Comment:
I guess we don't need the `finished = ` part of this, right? (Since we're not using it anywhere?)

Reply:
it's been a while but I think we need it, see this comment in the blog post I was inspired to tackle this (https://technology.amis.nl/2019/06/06/graceful-shutdown-of-forked-workers-in-python-and-javascript-running-in-docker-containers/): 

>  Do not forget to wait until the worker is finished with finished = os.waitpid(worker_pid, 0) or else the master might be finished before the worker causing the worker to be killed in a not so graceful matter.

Reply:
or even better we could replace it with what's above https://github.com/encode/uvicorn/pull/522#issuecomment-565047807 which would help debugging corner cases

Reply:
I think Tom meant we don't need to store the result in a variable — `os.waitpid(child_pid, 0)` should be enough.